test/unit/application_test.rb

```diff
@@ -356,7 +356,7 @@ class ApplicationTest < ActiveSupport::TestCase
       FactoryBot.create(:deployment, application: app, version: "v111", environment: "staging")
       FactoryBot.create(:deployment, application: app, version: "v111", environment: "integration")
 
-      assert_equal '{"integration"=>{"app_instance"=>"app1", "image"=>"v111", "created_at"=>"2025-01-29T14:27:01Z", "previous_version"=>nil, "github"=>""}, "staging"=>{"app_instance"=>"app1", "image"=>"v111", "created_at"=>"2025-01-29T14:27:01Z", "previous_version"=>nil, "github"=>""}}', app.current_image_deployed_by_environment.to_s
```

Comment:
Why this change? Would be helpful to explain in the commit. I'd imagine it's more useful to keep it as `nil` and then do any necessary casting of that value (e.g. to empty string) in the view rather than the model.